src/xcode/ENA/ENA/Source/HTTPClient/Service/Models/Receive/ModelWithCache.swift

```diff
@@ -0,0 +1,50 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+protocol ModelWithCaching {
+	var isCached: Bool { get set }
+}
+
+/**
+ Generel model to encapsulate the real model, which has to fullfill CBORDecoding. The main purpose of this model is to store a property if the real model returned from the service is from the cache or freshly fetched. Normally you cannot differentiate here.
+ */
+struct ModelWithCache<WrappedModel>: CBORDecoding, ModelWithCaching where WrappedModel: CBORDecoding {
```

Comment:
This solution would only works for CBORResources? Would it be possible to have a general approach?

Reply:
Yeah, good idea, but this would lead into an huge refactoring because the resource's decoding work not all the same way (JSON and Protobuf offer an decoding API, CBOR not). We would need to change our architecture here.
So i would like to let it as it is.

Reply:
And what if we would resolve this at a higher level. For example, would it be possible to have a func 'loadWithMetadata(...)' Which returns a touple, metadata and the model?

Reply:
I will have a look.

Reply:
Okay, on the level as new API call it is not as easy as it seems. But I refactored on the model layer some things.